Library/Homebrew/formula.rb

```diff
@@ -1013,9 +1018,10 @@ def _outdated_versions(options = {})
 
     installed_kegs.each do |keg|
       version = keg.version
+      tab = Tab.for_keg(keg)
       all_versions << version
 
-      return [] if pkg_version <= version && !version.head?
+      return [] if pkg_version <= version && epoch == tab.epoch && !version.head?
```

Comment:
`latest_version`, `same_epoch`, `not_head` perhaps (as I can't think of a way of decribing "not head" without "not").